include/Parser/src/JSONParser.cpp

```diff
@@ -0,0 +1,125 @@
+#include "JSONParser.h"
+
+JSON::JSON() : m_type(NIL) {}
+
+JSON::JSON(Type t) : m_type(t) {}
+
+JSON::JSON(const std::string &value) : m_type(STRING), m_str_value(value) {}
+
+JSON::JSON(double value) : m_type(NUMBER), m_num_value(value) {}
+
+JSON::JSON(bool value) : m_type(BOOL), m_bool_value(value) {}
+
+JSON::Type JSON::get_type() const { return m_type; }
+
+const std::unordered_map<std::string, JSON>& JSON::get_object_value() const { return m_object_value; }
+
+const std::string& JSON::get_string_value() const { return m_str_value; }
+
+double JSON::get_number_value() const { return m_num_value; }
+
+bool JSON::get_bool_value() const { return m_bool_value; }
+
+void JSON::SkipWhitespace(std::stringstream &ss) {
+    while (ss.peek() == ' ' || ss.peek() == '\n' || ss.peek() == '\t' || ss.peek() == '\r') {
+        ss.get();
+    }
+}
+
+
+JSON JSON::Parse(const std::string &content) {
+    std::stringstream ss(content);
+    SkipWhitespace(ss);
+    char c = ss.peek();
+    if (c == '{') {
+        return ParseObject(ss);
+    } else if (c == '"') {
+        return ParseString(ss);
+    } else if (isdigit(c) || c == '-') {
+        return ParseNumber(ss);
+    } else if (c == 't' || c == 'f') {
+        return ParseBool(ss);
+    } else if (c == 'n') {
+        return ParseNull(ss);
+    } else {
+        throw std::runtime_error("Invalid JSON input");
```

Comment:
Looks like mistake in the JSON config can potentially crash the code.
* runtime exception is not best choise here, could be manually defined parsing error type or exception
* it will be better to gracefully handle all parsing errors/exceptions 
* errors could be more descriptive than invalid JSON to help user to find a mistake